apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java

```diff
@@ -239,38 +272,84 @@ protected Optional<TableMetadata> mapTableResponseToTableMetadata(
 
     TableMetadata.TableMetadataBuilder<?, ?> builder =
         TableMetadata.builder()
-            .creator(tableResponseBody.getTableCreator())
-            .dbName(tableResponseBody.getDatabaseId())
-            .tableName(tableResponseBody.getTableId())
+            .creator(Objects.requireNonNull(tableResponseBody.getTableCreator()))
+            .dbName(Objects.requireNonNull(tableResponseBody.getDatabaseId()))
+            .tableName(Objects.requireNonNull(tableResponseBody.getTableId()))
             .isPrimary(
                 tableResponseBody.getTableType()
                     == GetTableResponseBody.TableTypeEnum.PRIMARY_TABLE)
             .isTimePartitioned(tableResponseBody.getTimePartitioning() != null)
             .isClustered(tableResponseBody.getClustering() != null)
-            .retentionConfig(getTableRetention(tableResponseBody).orElse(null));
-    if (tableResponseBody.getCreationTime() != null) {
-      builder.creationTimeMs(tableResponseBody.getCreationTime());
-    }
+            .retentionConfig(getTableRetention(tableResponseBody).orElse(null))
+            .jobExecutionProperties(getJobExecutionProperties(tableResponseBody));
+    builder.creationTimeMs(Objects.requireNonNull(tableResponseBody.getCreationTime()));
```

Comment:
My understanding is that it should never be null, are there scenarios when it can be null? I understand that it could be there for backward compatibility with old tables that don't have that attribute, I think we can fix those tables...